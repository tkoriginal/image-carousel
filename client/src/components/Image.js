import React from 'react';
import styled from 'styled-components';

// Image is set to background image of DIV for easier manipulation
// Can be modified in a future implementation

const Wrapper = styled.div`
  margin: 0;
  width: 800px;
  height: 600px;
  background: url(${props => props.image});
  background-size: cover;
  background-repeat: no-repeat;
  background-position: 50% 50%;
`;

function Image(props) {
  return <Wrapper image={props.image} />;
}

export default Image;
