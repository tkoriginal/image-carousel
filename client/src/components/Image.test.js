import { shallow } from 'enzyme';
import React from 'react';
import Image from './Image';

describe('Image test', () => {
  it('Snapshot test', () => {
    const wrapper = shallow(<Image image="http://www.google.com/" />);
    expect(wrapper).toMatchSnapshot();
  });
});
