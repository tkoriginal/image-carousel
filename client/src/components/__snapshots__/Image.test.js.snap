// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image test Snapshot test 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Image
    image="http://www.google.com/"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "image": "http://www.google.com/",
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  margin: 0;
  width: 800px;
  height: 600px;
  background: url(",
          [Function],
          ");
  background-size: cover;
  background-repeat: no-repeat;
  background-position: 50% 50%;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "image": "http://www.google.com/",
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
  margin: 0;
  width: 800px;
  height: 600px;
  background: url(",
            [Function],
            ");
  background-size: cover;
  background-repeat: no-repeat;
  background-position: 50% 50%;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
